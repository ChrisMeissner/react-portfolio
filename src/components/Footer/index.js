import React from 'react';
import githubIcon from '../../assets/pics/githubIcon.png';
import instagramIcon from '../../assets/pics/instagramIcon.png';
import linkedInIcon from '../../assets/pics/linkedInIcon.png'

function Footer() {

  const footerContainerStyle = {
    'display': 'flex',
    'flex-direction': 'row',
    'justify-content': 'space-between',
    'align-items': 'center',
    'width': '30%',
    'margin': 'auto'
  };

  const anchorStyle = {
    'display': 'flex',
    'flex-direction': 'row',
    'justify-content': "center",
    'width': '100%',
    'margin': '2.5%'
  }

  const iconStyle = {
    'display': 'flex',
    'flex-direction': 'row',
    'justify-content': 'center',
    'width': 'inherit'
  };

  return (
    <div style={footerContainerStyle}>
      <a href="https://github.com/ChrisMeissner" style={anchorStyle}>
        <img src={githubIcon} alt="github-icon" style={iconStyle}/>
      </a>
      <a href="https://www.linkedin.com/in/chris-meissner-a8a331149/" style={anchorStyle}>
        <img src={linkedInIcon} alt="linkedIn-icon" style={iconStyle}/>
      </a>
      <a href="https://www.instagram.com/chris_meissner/?hl=en" style={anchorStyle}>
        <img src={instagramIcon} alt="instagram-icon" style={iconStyle}/>
      </a>
    </div>
  )
}

export default Footer;