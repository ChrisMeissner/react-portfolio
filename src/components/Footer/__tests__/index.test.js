import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Footer from '..';

afterEach(cleanup);

describe('Footer Component', () => {
  // renders Footer test
  // first test
  it('renders', () => {
    render(<Footer/>);
  });

  // second test
  it('matches snapshot DOM node structure', () => {
    // render Footer
    const { asFragment } = render(<Footer />);
    expect(asFragment()).toMatchSnapshot();
  })

});