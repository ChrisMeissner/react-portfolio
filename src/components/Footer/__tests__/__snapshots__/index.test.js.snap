// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <section
    style="display: flex; flex-direction: row; justify-content: space-between; align-items: center; width: 30%; margin: auto; background-color: rgb(255, 41, 91); border-radius: 15px;"
  >
    <a
      href="https://github.com/ChrisMeissner"
      style="display: flex; flex-direction: row; justify-content: center; width: 100%; margin: 2.5%;"
    >
      <img
        alt="github-icon"
        src="githubIcon.png"
        style="display: flex; flex-direction: row; justify-content: center; width: inherit;"
      />
    </a>
    <a
      href="https://www.linkedin.com/in/chris-meissner-a8a331149/"
      style="display: flex; flex-direction: row; justify-content: center; width: 100%; margin: 2.5%;"
    >
      <img
        alt="linkedIn-icon"
        src="linkedInIcon.png"
        style="display: flex; flex-direction: row; justify-content: center; width: inherit;"
      />
    </a>
    <a
      href="https://www.instagram.com/chris_meissner/?hl=en"
      style="display: flex; flex-direction: row; justify-content: center; width: 100%; margin: 2.5%;"
    >
      <img
        alt="instagram-icon"
        src="instagramIcon.png"
        style="display: flex; flex-direction: row; justify-content: center; width: inherit;"
      />
    </a>
  </section>
</DocumentFragment>
`;
